### PR TITLE
[feat] implement remove plant functionality & style PlantPages

### LINKED ISSUE #49: Style PlantPage and Add Remove Plant Functionality
https://linear.app/trap-garden/issue/TG-42/style-plantpage-and-add-remove-plant-functionality

remember to git checkout main → git pull → git checkout -b [thisbranch]
* assigned: Monday 11/18
* due (PR open by): Sunday 11/24 
* MERGE by: Sunday 12/1

### PR DESCRIPTION
**Pull request by SashankBalusu**
### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- styled the general and user plant page 
   - consolidates styling to /plant-pages/styles.ts. Both Plant Pags are also under the `plant-page` directory. 
- allow user to remove plant from their plants
   - this actually means we update the `date_removed`; we don't actually delete the row from the db 
- also made edits to the plantpage components in order to match the styles
- updated `UserPlant` type to include date_removed rather than date_harvested. also, planting_type is now PlantingTypeEnum type.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy! Use the provided image template. Drag the desired image into the PR, then copy the link into the placeholder."
Previously: 
Screenshot 2024-11-22 at 1 51 10 AM
Screenshot 2024-11-22 at 1 50 42 AM
Note that the images of plants were hard coded for proof of concept, if you were to pull and run you will not see the image

Now: 
GeneralPlantPage
Screen Shot 2024-12-07 at 1 46 33 AM

UserPlantPage 
Screen Shot 2024-12-07 at 1 46 45 AM

## How to review
[//]: # 'Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?'

- can view both the general and user plant pages, test back button and remove button

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."
* create the SeasonColorKey and use it in the PlantPages (maybe create a new component, `PlantingTimeline`, since both PlantPages use it)
* on the issue of whether we should actually delete rows when "Removing Plants": POC wants a Retool workflow to filter out plants that have been removed a season ago. 
* fix /add-details handling of "SELECT" value, which is now invalid as a `planting_type`

### Online sources
[//]: # 'Copy links to any tutorials or documentation that was useful to you when working on this PR'

### Related PRs
[//]: # "Add related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

CC: @ccatherinetan

### CODE REVIEW

File: api/supabase/queries/userPlants.ts [hunk 1]
```diff
@@ -44,3 +44,15 @@ export async function getCurrentUserPlantsByUserId(
...
+export async function removeUserPlantById(id: UUID) {
+  const { data, error } = await supabase
+    .from('user_plants')
+    .delete()
+    .eq('id', id);
+
+  if (error) {
+    console.error('Error deleting row:', error);
+    return null;
+  }
```
Comment (ccatherinetan) [inline api/supabase/queries/userPlants.ts]: note: honestly i'm down to generally throw errors in queries, but that would require us to catch the error wherever the query is used. this allows us to use the error message + do more specific/custom error-handling. this is fine for now, but if we do want to do custom error-handling, we might consider changing it to throwing an error later

File: app/all-plants/[plantId]/page.tsx [hunk 2]
```diff
@@ -18,12 +36,52 @@ export default function GeneralPlantPage() {
...
+        <>
+          <Header>
+            <HeaderContent>
+              <ButtonWrapper>
+                <BackButton
+                  onClick={() => {
+                    router.push(`/view-plants`);
+                  }}
+                >
+                  ←
```
Comment (ccatherinetan) [inline app/all-plants/[plantId]/page.tsx]: note: this is good for now! we might want to export the arrow icon from figma eventually??

File: app/all-plants/[plantId]/style.ts [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+`;
+export const AddPlant = styled.button`
+  background-color: ${COLORS.shrub};
+  color: white;
+  padding: 8px 16px;
+  border: none;
+  border-radius: 5px;
+  cursor: pointer;
+  font-size: 0.75rem;
+  font-style: inherit;
```
Comment (ccatherinetan) [inline app/all-plants/[plantId]/style.ts]: styles are inherited by default so we can remove this line!

File: components/YourPlantDetails/style.ts [hunk 4]
```diff
@@ -33,10 +37,21 @@ export const DetailRow = styled.div`
...
 `;
 
 export const StyledIcon = styled.div`
   color: ${COLORS.shrub};
   display: flex;
   align-items: center;
 `;
+
+export const StyledHeading = styled(H3)`
+  font-size: 16px;
```
Comment (ccatherinetan) [inline components/YourPlantDetails/style.ts]: font sizes should be in rem! 16px = 1 rem

Also, note: if this is a H3, it should inherit the sizing from H3, so you wouldn't need to re-define it here. Kyrene is working on making everything consistent, so you can keep the font-size for now

File: app/all-plants/[plantId]/page.tsx [hunk 5]
```diff
@@ -18,12 +36,52 @@ export default function GeneralPlantPage() {
     getPlant();
   }, [plantId]);
   return (
-    <div>
+    <Container>
       {currentPlant && (
```
Comment (ccatherinetan) [inline app/all-plants/[plantId]/page.tsx]: note: maybe we should add a spinner if it's loading in the future? let's discuss in worksesh

File: app/all-plants/[plantId]/page.tsx [hunk 6]
```diff
@@ -18,12 +36,52 @@ export default function GeneralPlantPage() {
...
+                <BackButton
+                  onClick={() => {
+                    router.push(`/view-plants`);
+                  }}
+                >
+                  ←
+                </BackButton>
+              </ButtonWrapper>
+              <PlantImage
+                src={currentPlant.img}
```
Comment (ccatherinetan) [inline app/all-plants/[plantId]/page.tsx]: note: eventually, we'd want to have a ternary statement here to provide the default image/icon if `plant.img` is null, i.e.
`currentPlant.img ? <PlantImage /> : <Icon type="defaultPlant" />`
but we can leave it for now

File: app/my-garden/[userPlantId]/page.tsx [hunk 7]
```diff
@@ -21,14 +44,65 @@ export default function UserPlantPage() {
     };
     getPlant();
   }, [userPlantId]);
+  async function removePlant() {
```
Comment (ccatherinetan) [inline app/my-garden/[userPlantId]/page.tsx]: note: eventually, we'd want to have some sort of warning to prevent users from accidentally removing plants

File: components/PlantCareDescription/index.tsx
Comment (ccatherinetan) [inline components/PlantCareDescription/index.tsx:1]: note: when kyrene makes the fonts consistent with the design system, we want all text styles to inherit from `styles/text.ts`, i.e. we should always have `styled(H2)` instead of a `styled.h2`. And if you need to use the component for a different html tag, you can use the `as` prop like so:
`<CustomStyle as="span">`. [Documentation](https://styled-components.com/docs/api) for as

File: app/all-plants/[plantId]/style.ts [hunk 8]
```diff
@@ -0,0 +1,81 @@
+import styled from 'styled-components';
+import COLORS from '@/styles/colors';
+
+export const Container = styled.div`
+  padding: 20px;
+`;
+
+export const Header = styled.div`
+  background-color: ${COLORS.backgroundGrey};
+  margin: -28px -28px 24px -28px;
```
Comment (ccatherinetan) [inline app/all-plants/[plantId]/style.ts]: nit: negative margin LOL; Although this is supported, unless we are explicitly trying to overlap elements (which doesn't seem to be the case here), we should try to avoid using negative margins. 

If this was done to remove the white space, this should be resolved by the global styles that sets margin and padding to 0px by default for all components. I will commit the global styles to main, but this can be simplified to 
```
export const Header = styled.div`
  background-color: ${COLORS.backgroundGrey};
  height: 220px; 
`;
```
Perhaps for consistency across the pages, height can be hardcoded to 220px? otherwise, perhaps this can be a min-height attribute?

(also see my comment about removing Container)

File: app/all-plants/[plantId]/style.ts [hunk 9]
```diff
@@ -0,0 +1,81 @@
...
+  font-style: normal;
+  font-weight: 400;
+  line-height: normal;
+  margin: 0;
+`;
+
+export const TitleWrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+  gap: 2px;
```
Comment (ccatherinetan) [inline app/all-plants/[plantId]/style.ts]: is this necessary? can we remove the gap?

File: app/my-garden/[userPlantId]/style.ts
Comment (ccatherinetan) [inline app/my-garden/[userPlantId]/style.ts:1]: nit: styling folders should be plural: this file should be renamed `styles.ts`

File: app/all-plants/[plantId]/style.ts [hunk 10]
```diff
@@ -0,0 +1,81 @@
+import styled from 'styled-components';
+import COLORS from '@/styles/colors';
+
+export const Container = styled.div`
+  padding: 20px;
+`;
```
Comment (ccatherinetan) [inline app/all-plants/[plantId]/style.ts]: we should remove the padding from Container and instead include `margin: 24px;` in `Content`

File: app/all-plants/[plantId]/style.ts [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+
+export const Content = styled.div`
+  display: flex;
+  flex-direction: column;
+  gap: 20px;
+`;
+
+export const NameWrapper = styled.div`
+  display: flex;
+  justify-content: flex-start;
```
Comment (ccatherinetan) [inline app/all-plants/[plantId]/style.ts]: nit: i think this is the default so we don't have to include this line here

File: app/all-plants/[plantId]/style.ts [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+  margin: 0;
+`;
+
+export const TitleWrapper = styled.div`
+  display: flex;
+  flex-direction: row;
+  gap: 2px;
+  justify-content: space-between;
+`;
+export const AddPlant = styled.button`
```
Comment (ccatherinetan) [inline app/all-plants/[plantId]/style.ts]: note: kyrene said this button style may change, so let's just leave this for now, and i'll update it later

File: app/my-garden/[userPlantId]/style.ts
Comment (ccatherinetan) [reply app/my-garden/[userPlantId]/style.ts:1]: also, since there are so many shared styles between GeneralPlantPage and UserPlantPage, we should just have them share styles rather than duplicate the styles, as this will reduce redundancy make it easier for us to keep styling consistent between pages. 

One way to do is this is to restructure the directories like so: 

UserPlantPage → `app/view-plants/my-garden/[userPlantId]/page.tsx`
- `[userPlantId]/styles.ts` if there are any custom styles that aren't shared with GeneralPlantPage

GeneralPlantPage → `app/view-plants/all-plants/[plantId]/page.tsx`
- `[plantId]/styles.ts` if there are any custom styles that aren't shared with userPlantPage

Shared Styles → `app/view-plants/shared/styles.ts`
- should include Container, Header, HeaderContainer, PlantImage, Content, NameWrapper etc.

**Review by ccatherinetan — COMMENTED** [review]
looking great, tysm sashank! a couple comments
1. nit: all styling files should be plural, i.e. `styles.ts` rather than `style.ts`
2. we might want to consolidate the shared styles between UserPlantPage and GeneralPlantPage. one way to do this is to move both components under the view-plants route and add a `shared/styles.ts` file (see my comment)
3. let's avoid using negative margins if it's not for overlapping components. I will push some global styles that will address the white space issue; see my comment about changing styles for Container, Header, and Content.

File: app/all-plants/[plantId]/style.ts (hunk 8 above)
Comment (ccatherinetan) [reply app/all-plants/[plantId]/style.ts]: update: i've merged in the global styles in main, so if u rebase u can see what i'm saying (i.e. no need to have negative margin)

File: api/supabase/queries/userPlants.ts [hunk 13]
```diff
@@ -44,3 +44,15 @@ export async function getCurrentUserPlantsByUserId(
   }
   return data;
 }
+export async function removeUserPlantById(id: UUID) {
```
Comment (ccatherinetan) [inline api/supabase/queries/userPlants.ts:66]: omg i just realized that we actually don't want to delete the plants from their lists, but instead include date_removed, so that it is no longer fetched
